programs/mango-v4/src/accounts_ix/openbook_v2_cancel_all_orders.rs

```diff
@@ -0,0 +1,47 @@
+use anchor_lang::prelude::*;
+
+use crate::error::*;
+use crate::state::*;
+use openbook_v2::{
+    program::OpenbookV2,
+    state::{BookSide, Market, OpenOrdersAccount},
+};
+
+#[derive(Accounts)]
+pub struct OpenbookV2CancelAllOrders<'info> {
+    #[account(
+        constraint = group.load()?.is_ix_enabled(IxGate::OpenbookV2CancelAllOrders) @ MangoError::IxIsDisabled,
+    )]
+    pub group: AccountLoader<'info, Group>,
+
+    #[account(
+        has_one = group,
+        constraint = account.load()?.is_operational() @ MangoError::AccountIsFrozen
+        // owner is checked at #1
+    )]
+    pub account: AccountLoader<'info, MangoAccountFixed>,
+    pub owner: Signer<'info>,
+
+    #[account(mut)]
+    pub open_orders: AccountLoader<'info, OpenOrdersAccount>,
+
+    #[account(
+        has_one = group,
+        has_one = openbook_v2_program,
+        has_one = openbook_v2_market_external,
+    )]
+    pub openbook_v2_market: AccountLoader<'info, OpenbookV2Market>,
+
+    pub openbook_v2_program: Program<'info, OpenbookV2>,
+    #[account(mut)]
+    pub openbook_v2_market_external: AccountLoader<'info, Market>,
+
+    #[account(mut)]
```

Comment:
Add a note saying that events/bids/asks accounts are just forwarded to a ob_v2 cpi call, and are validated there.

Reply:
I added some constraints by loading the open-book market and check if it corresponds to asks, bids and event queue.